Platform/AlderlakeBoardPkg/Library/FspmUpdUpdateLib/FspmUpdUpdateLib.c

```diff
@@ -266,33 +269,54 @@ UpdateFspConfig (
+    SpdData[4] = (UINT32)(UINTN) (((MEM_SPD3_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD3_TAG))->MemorySpdPtr3);
+    SpdData[5] = (UINT32)(UINTN) (((MEM_SPD4_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD4_TAG))->MemorySpdPtr4);
+    SpdData[6] = (UINT32)(UINTN) (((MEM_SPD5_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD5_TAG))->MemorySpdPtr5);
+    SpdData[7] = (UINT32)(UINTN) (((MEM_SPD6_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD6_TAG))->MemorySpdPtr6);
+    SpdData[8] = (UINT32)(UINTN) (((MEM_SPD7_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD7_TAG))->MemorySpdPtr7);
+#if defined(PLATFORM_RPLA)
   SpdData[9] = (UINT32)(UINTN) (((MEM_SPD8_CFG_DATA *)FindConfigDataByTag (CDATA_MEM_SPD8_TAG))->MemorySpdPtr8);
-  Fspmcfg->MemorySpdPtr000  = SpdData[MemCfgData->SpdDataSel000];
-  Fspmcfg->MemorySpdPtr001  = SpdData[MemCfgData->SpdDataSel001];
-  Fspmcfg->MemorySpdPtr010  = SpdData[MemCfgData->SpdDataSel010];
-  Fspmcfg->MemorySpdPtr011  = SpdData[MemCfgData->SpdDataSel011];
-  Fspmcfg->MemorySpdPtr020  = SpdData[MemCfgData->SpdDataSel020];
-  Fspmcfg->MemorySpdPtr021  = SpdData[MemCfgData->SpdDataSel021];
-  Fspmcfg->MemorySpdPtr030  = SpdData[MemCfgData->SpdDataSel030];
-  Fspmcfg->MemorySpdPtr031  = SpdData[MemCfgData->SpdDataSel031];
-  Fspmcfg->MemorySpdPtr100  = SpdData[MemCfgData->SpdDataSel100];
-  Fspmcfg->MemorySpdPtr101  = SpdData[MemCfgData->SpdDataSel101];
-  Fspmcfg->MemorySpdPtr110  = SpdData[MemCfgData->SpdDataSel110];
-  Fspmcfg->MemorySpdPtr111  = SpdData[MemCfgData->SpdDataSel111];
-  Fspmcfg->MemorySpdPtr120  = SpdData[MemCfgData->SpdDataSel120];
-  Fspmcfg->MemorySpdPtr121  = SpdData[MemCfgData->SpdDataSel121];
-  Fspmcfg->MemorySpdPtr130  = SpdData[MemCfgData->SpdDataSel130];
-  Fspmcfg->MemorySpdPtr131  = SpdData[MemCfgData->SpdDataSel131];
+#endif
+    Fspmcfg->MemorySpdPtr000  = SpdData[MemCfgData->SpdDataSel000];
+    Fspmcfg->MemorySpdPtr001  = SpdData[MemCfgData->SpdDataSel001];
+    Fspmcfg->MemorySpdPtr010  = SpdData[MemCfgData->SpdDataSel010];
+    Fspmcfg->MemorySpdPtr011  = SpdData[MemCfgData->SpdDataSel011];
+    Fspmcfg->MemorySpdPtr020  = SpdData[MemCfgData->SpdDataSel020];
+    Fspmcfg->MemorySpdPtr021  = SpdData[MemCfgData->SpdDataSel021];
+    Fspmcfg->MemorySpdPtr030  = SpdData[MemCfgData->SpdDataSel030];
+    Fspmcfg->MemorySpdPtr031  = SpdData[MemCfgData->SpdDataSel031];
+    Fspmcfg->MemorySpdPtr100  = SpdData[MemCfgData->SpdDataSel100];
+    Fspmcfg->MemorySpdPtr101  = SpdData[MemCfgData->SpdDataSel101];
+    Fspmcfg->MemorySpdPtr110  = SpdData[MemCfgData->SpdDataSel110];
+    Fspmcfg->MemorySpdPtr111  = SpdData[MemCfgData->SpdDataSel111];
+    Fspmcfg->MemorySpdPtr120  = SpdData[MemCfgData->SpdDataSel120];
+    Fspmcfg->MemorySpdPtr121  = SpdData[MemCfgData->SpdDataSel121];
+    Fspmcfg->MemorySpdPtr130  = SpdData[MemCfgData->SpdDataSel130];
+    Fspmcfg->MemorySpdPtr131  = SpdData[MemCfgData->SpdDataSel131];
+  }
```

Comment:
It is preferred to use SBL config data for SPD data. It doesn't need change code when changing the memory, and it doesn't impact other platforms.
For your use case, often it needs different PlatformId for different memory configuration.  Given the only difference is the memory, I think it is ok to use a single PlatformId and update MemCfgData->SpdDataSel dynamically using MemId. This way you just need minimal code change here.

Reply:
@gdong1 there are two things here that are not matching:

1. The order of the SpdData arrays in SpdTableRplpLp5Auto is different then the way Sbl loads the SpdData.bin files (other order, other memories supported) so even though we update the MemCfgData->SpdDataSelXXX with MemId this would not have the same effect. 
2. Our automotive configuration could have same PlatformId but different memory configurations e.g. 16GB/32GB/64Gb - and this is configured by Manufacturing in the FRU chip.

Reply:
I didn't get your point on the matching or order issue.
It looks you have 9 SPD data blobs for this auto platform, please check if this flow works.
1. In save the SPD data into Files
     e.g.: save mRplpLp5AutoMicron4GMT62F1G32D2DSSpd into Spd_Lp5_Auto_Micron_4G.bin
2. Update auto platform dlt file, override MEMORY_CFG_DATA.MemorySpdPtrx as below
    MEMORY_CFG_DATA.MemorySpdPtr0  | {FILE: Spd_Lp5_Auto_Micron_4G.bin}
    MEMORY_CFG_DATA.MemorySpdPtr1  | {FILE: Spd_Lp5_Auto_Micron_8G.bin}
    MEMORY_CFG_DATA.MemorySpdPtr2  | {FILE: Spd_Lp5_Auto_Micron_16G.bin}
    ...
    MEMORY_CFG_DATA.MemorySpdPtr8  | {FILE: Spd_Lp5_Auto_Samsung_16G.bin}

  # override the default value
  MEMORY_CFG_DATA.SpdDataSel101            | 0

3. Just need add the logic as below in this file 
 
  #if defined(PLATFORM_RPLA)
   MemId = PcdGet8 (PcdMemConfigId);
    Fspmcfg->MemorySpdPtr000  = MemId;
    Fspmcfg->MemorySpdPtr010  = MemId;
    Fspmcfg->MemorySpdPtr020  = MemId;
    Fspmcfg->MemorySpdPtr030  = MemId;
    Fspmcfg->MemorySpdPtr100  = MemId;
    Fspmcfg->MemorySpdPtr110  = MemId;
    Fspmcfg->MemorySpdPtr120  = MemId;
    Fspmcfg->MemorySpdPtr130  = MemId;
#endif

Please specify the exact place if have issue with above update.

Reply:
Hi @gdong1 ! thanks a lot for the suggestion! However, wouldn't the proposed solution break the current implementation, considering the fact that we will overwrite the SpdPtr? e.g.

MEMORY_CFG_DATA.MemorySpdPtr0 | {FILE: Spd_Lp5_Auto_Micron_4G.bin}
MEMORY_CFG_DATA.MemorySpdPtr1 | {FILE: Spd_Lp5_Auto_Micron_8G.bin}
MEMORY_CFG_DATA.MemorySpdPtr2 | {FILE: Spd_Lp5_Auto_Micron_16G.bin}
...
MEMORY_CFG_DATA.MemorySpdPtr8 | {FILE: Spd_Lp5_Auto_Samsung_16G.bin}

This will break, at least in theory, the existing support for the already existing memory configurations that Rplp supports. However, since we don't need any of those memory configurations, and since we are maintaining our own SblInternal/Platform/RplAuto/CfgData/*** i guess i wouldn't be a problem to remove all the other memories and simply add support for our own.

Reply:
All the changes in DLT file will only apply to the specific platform, there is no impact to other platforms.  If you find it doesn't work, it should be a bug.

Reply:
Done.
Thanks for the suggestion.